### PR TITLE
Add verbose option to the CLI, and general improvements to local logging

### LINKED ISSUE #144: Enable logging for the BPA
The BPA should log any errors encountered during a run.

### PR DESCRIPTION
**Pull request by VeraBE**
## Description

Closes #144 

Working towards #157 too

Documentation followed: https://docs.microsoft.com/en-us/dotnet/core/extensions/logging

Changes in the CLI are not ideal, we need to do #207 

I'm not sure if we want to replace all the remaining `Console.WriteLine`s for `logger.LogInformation`, I assumed we don't

I think ideally we should pass the logger to `TemplateAnalyzer` and the engines on class creation, but the `TemplateAnalyzer` instance is created before processing the CLI input options, and I'm not sure if we want to change that now

Regarding tests, #28 has to be done before we can add proper tests for this PR

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [x] The pull request does not introduce breaking changes.

### [General Guidelines](../CONTRIBUTING.md)
- [x] Title of the pull request is clear and informative.
- [x] Description of the pull request is clear and informative.
- [x] I have added myself to the 'assignees'.
- [x] I have added 'linked issues' if relevant.

### [Testing Guidelines](../CONTRIBUTING.md#tests)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by anaismiller — APPROVED** [review]

File: src/Analyzer.Cli/CommandLineParser.cs [hunk 1]
```diff
@@ -285,5 +301,22 @@ private IReportWriter GetReportWriter(string reportFormat, FileInfo outputFile,
...
+            var logLevel = verbose ? LogLevel.Debug : LogLevel.Information;
+
+            using var loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder
+                .SetMinimumLevel(logLevel)
+                .AddSimpleConsole(options =>
+                {
+                    options.SingleLine = true;
+                });
```
Comment (JohnathonMohr) [inline src/Analyzer.Cli/CommandLineParser.cs:316]: Indent these lines.

File: src/Analyzer.TemplateProcessor/ArmTemplateProcessor.cs [hunk 2]
```diff
@@ -382,9 +390,12 @@ private void ProcessTemplateResourceLanguageExpressions(TemplateResource templat
                         evaluationContext: evaluationHelper.EvaluationContext);
                 }
             }
-            catch
+            catch (Exception ex)
             {
                 // Do not throw if there was an issue with evaluating language expressions
+
+                logger?.LogError(ex, "An exception occurred while evaluating the properties of a resource. Properties: {properties}", templateResource?.Properties?.Value);
```
Comment (JohnathonMohr) [inline src/Analyzer.TemplateProcessor/ArmTemplateProcessor.cs:397]: minor: null-coalescing operator isn't needed here, because if `templateResource` or `templateResource.Properties` was null, nothing would happen in the `try` block.

File: src/Analyzer.TemplateProcessor/ArmTemplateProcessor.cs [hunk 3]
```diff
@@ -185,6 +191,8 @@ internal Template ProcessResourcesAndOutputs(Template template)
                     }
                     catch (Exception)
                     {
+                        logger?.LogError("The parsing of a template output failed. Output key: {outputKey}. Root value: {rootValue}", outputKey, template.Outputs[outputKey]?.Value?.Value);
```
Comment (JohnathonMohr) [inline src/Analyzer.TemplateProcessor/ArmTemplateProcessor.cs:194]: Maybe reword to "... Output name: {outputKey}. Output value: {rootValue}".

File: src/Analyzer.Cli/CommandLineParser.cs [hunk 4]
```diff
@@ -75,15 +76,21 @@ private RootCommand SetupCommandLineAPI()
             outputFileOption.AddAlias("-o");
             analyzeTemplateCommand.AddOption(outputFileOption);
 
+            var verboseOption = new Option(
```
Comment (JohnathonMohr) [inline src/Analyzer.Cli/CommandLineParser.cs:79]: Add this to README

**Review by JohnathonMohr — CHANGES_REQUESTED** [review]
Just a few minor changes, mostly the README update.

**Review by JohnathonMohr — APPROVED** [review]
